package day0303;

import java.util.Random;

/**
 *	예외 강제발생
 * @author user
 */
public class UseThrow {
	/**
	 * 발생된 예외는 대부분은 던져서 처리한다
	 * 던지는 예외는 발생된 예외를 그대로 던져도 되고, 부모예외처리 클래스를 사용하여 던져도 된다
	 */
	public void useThrow() throws Exception {
		System.out.println("예외발생 전");
		if( new Random().nextBoolean() ) {
			//예외를 강제로 발생시킬려면 :throw new 예외처리클래스();
//			try {
				throw new Exception("예외생산함!!");
//			}catch(Exception e) {
//				// 발생한 예외를 바로 try ~catch로 처리할 수 있다
//				e.printStackTrace();
//			}//end catch
		}//end if
		System.out.println("예외발생 후");
		
	}//useThrow

	public void callUseThrow() {
		try {
			useThrow();
		}catch(Exception e) {
			System.err.println("호출한 곳에서 예외 처리"+e);
		}//end catch
	}//callUseThrow
	
	public static void main(String[] args) {
		UseThrow ut = new UseThrow();
		ut.callUseThrow();
	}//main

}//class
